### ISSUE TITLE
rvz, cso gets ignored doing import content

### ISSUE THREAD

**Issue by beyondmeat** [libretro/RetroArch#13735]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

I have gamecube and wii RVZ files and PSP cso files that the scanner ignores and does not make playlists for them.

### Expected behavior
It works like any other supported format
[What you expected to happen]

### Actual behavior

it scans but no playlist gets made, content not added to respective playlists. both dolphin and ppsspp cores installed
### Steps to reproduce the bug

1. put rvz compressed gamecube games in roms/ngc, following redump naming
2. [import content, scan directory...
3. no playlists made, repeat for psp cso roms
### Bisect Results

n/a

### Version/Commit
You can find this information under Information/System Information

- RetroArch: [version/commit]
nightly 3/10/22

also same with lakka

### Environment information

- OS: lakka on rpi400, win11
- 

shouldn't need to force with manual scan but worked around it

**Comment by prestidigitonium** [comment]
I've the same problem, .rvz are not recognized (I'm on archlinux with the latest retroarch stable) and dolphin core is correctly installed

**Comment by Ric-82** [comment]
I'm having the same problem. From what I understand, it's because to create a playlist consistent with the reference database, RetroArch needs to match the disc's serial number, but apparently it can't read the disc's serial number if the disc is in rvz format. In fact, the reference database file used by RetroArch contains games in ISO format. I've done various tests, and unfortunately (as explained here: https://github.com/libretro/libretro-database#fields--headers) large disc-based games necessarily require a serial number match. A matching CRC isn't enough.

I don't know if the rvz format itself is impossible to scan for this type of checksum (since the serial number is encoded in binary data) or if this is a limitation specific to RetroArch, in which case I hope it will be addressed in future updates.

**Comment by Jamiras** [comment]
The general problem is that .rvz and .cso are compressed formats (similar to .chd) and require dedicated code to decode them. The helper library for .chd has been incorporated into RetroArch, so it can read chd files. The appropriate code for .rvz and .cso hasn't been, so RetroArch doesn't know where to find the serials.

**Comment by Ric-82** [comment]
Yes, I imagined it was something like that. Hopefully, full support for .rvz files will be implemented in the future, as has already been done for .chd files.

In the meantime, a solution to "force" the GameCube game database matches is to create a corresponding .m3u file for each .rvz file (it seems tedious, but it can be automated via script. I asked Copilot to create it, and it worked great), then edit the database's .rdb file, replacing the .iso ROM names with those in .m3u format.

To do this, I recommend using the HashMyFiles tool (just Google it and you'll find it quickly) to get a list of newly created .m3u files with their sizes, CRC32, MD5, and SHA1. Next, copy and paste this information into an Excel or OpenOffice spreadsheet and use it in a new .rdb file created with the RDBEd tool (also easily found on Google). We then need to compare the information in our rdb with that of the original RetroArch database to obtain the correct game names (this is done using RDBEd's "import file" command).

After making the necessary changes and adjustments, and placing the newly created rdb file in the RetroArch folder with the other rdbs (overwriting the old one), by running the automatic scan, we should be able to correctly view our collection in the front end.

It seems complicated, but it's more tedious than difficult. Unfortunately, RetroArch's stringency in finding exact matches in the databases requires a bit of extra work.

**Comment by warmenhoven** [comment]
Scanning rvz files works fine for me. Pulling the serial out of the rvz got added in 1.11. Unfortunately I don't know that there's enough information in the current logging to adequately debug the issue further. I'll plan to improve the logging to make it easier to see what's failing, and whether it's reading the serial wrong or whether it just isn't matching the redump db that's used.
